prover/utils/arena/vector_arena.go

```diff
@@ -50,9 +50,21 @@ func (a *VectorArena) Offset() int64 {
 	return atomic.LoadInt64(&a.offset)
 }
 
+// Remaining returns the number of elements of type T that can still be allocated
+// from the arena before it is exhausted.
+func Remaining[T any](a *VectorArena) int {
```

Comment:
I guess you already tried, but generically there is also Go experiment for arenas https://go.dev/src/arena/arena.go

I think implementation-wise it is similar, but I understand GC knows arenas better so for example if you would do experiments Arena.Free then memory can be reused without running GC.

We could have two implementations as is and hide the experiment arena behind the `goexperiment.arena` build tag.